src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/CompositeModelBinder.cs

```diff
@@ -36,70 +36,77 @@ public CompositeModelBinder(IList<IModelBinder> modelBinders)
         /// <inheritdoc />
         public IList<IModelBinder> ModelBinders { get; }
 
-        public virtual Task<ModelBindingResult> BindModelAsync(ModelBindingContext bindingContext)
+        public virtual Task BindModelAsync(ModelBindingContext bindingContext)
         {
             if (bindingContext == null)
             {
                 throw new ArgumentNullException(nameof(bindingContext));
             }
 
-            var newBindingContext = CreateNewBindingContext(bindingContext);
-            if (newBindingContext == null)
-            {
-                // Unable to find a value provider for this binding source. Binding will fail.
-                return ModelBindingResult.NoResultAsync;
-            }
-
-            return RunModelBinders(newBindingContext);
+            return RunModelBinders(bindingContext);
```

Comment:
No longer any value in having `RunModelBinders()` separate. Might not be a big deal since the old `CreateNewBindingContext()` rarely failed. But we should measure a couple of scenarios to be sure we're not losing perf w/o the previous optimization here.